### PR TITLE
Add GPU support take 3

### PR DESCRIPTION
**Pull request by ablaom**
Same as #66 but with squashed commits

### CODE REVIEW

**Comment by ablaom** [comment]
@ayush-1506 Would you be able to review this?

cc @OkonSamuel

**Comment by ayush-1506** [comment]
@ablaom Sure, will take a look at this later today.

File: src/image.jl [hunk 1]
```diff
@@ -11,34 +11,44 @@ mutable struct ImageClassifier{B,F,O,L} <: MLJModelInterface.Probabilistic
...
+    
+function MLJModelInterface.fit(model::ImageClassifier,
+                               verbosity::Int,
+                               X_,
+                               y_)
 
     data = collate(model, X_, y_)
 
-    levels = y_ |> first |> MLJModelInterface.classes
+    levels =     levels = MLJModelInterface.classes(y_[1])
```
Comment (ayush-1506) [inline src/image.jl:51]: Looks like a typo.

**Review by ayush-1506 — COMMENTED** [review]
Apart from one small typo, everything looks great. Eager to try this out.

File: src/image.jl (hunk 1 above)
Comment (ablaom) [reply src/image.jl:51]: I think it's the same but your suggestion more readable.

**Comment by ablaom** [comment]
@ayush-1506 thanks for the review!

File: src/image.jl (hunk 1 above)
Comment (ablaom) [reply src/image.jl:51]: Yes.

**Pull request merged**